### DISCUSSION TITLE
develop branch

### DISCUSSION THREAD

**Discussion by rocketman768** [Brewtarget/brewtarget#610] [Q&A]
@Brewtarget/developers 

Over the last few months, we've been moving to the [standard git branching model](http://nvie.com/posts/a-successful-git-branching-model/). As one of the final parts of the process, I created and made the 'develop' branch the default branch. All you need to do is treat it exactly the same way we have been treating the 'master' branch. Please base all your new features on it, and create pull requests against it.

On and after the 2.4.0 release this summer, the 'master' branch will be used to track releases only.

Thank you!

**Comment by theophae** [comment]
Do you plane to move to continuous integration process with automatic merge into develop branch based on Travis result ?
About Travis tools, it could be a good idea to make it execute brewtarget_tests automatically for every pull request ?

I've also read some interesting things about [fast-forwarding in git](https://sandofsky.com/images/fast_forward.pdf). Base on that, I think we could avoid squashing commits systematically for the pull request and use `--no-ff` option for git merge instead. What do you think about this ?

**Comment by rocketman768** [comment]
Yes, I'd like to execute brewtarget_tests as part of the travis builds. If you know how, submit a pull request.

I would not like to do any automatic merging for now.

But, I would really REALLY like automatic packaging for:
- debian/stable (x86, amd64)
- debian/unstable (x86, amd64)
- fedora (x86, amd64)
- suse (x86, amd64)
- Windows (x86)
- Mac (amd64)

**Comment by theophae** [comment]
About the tests, it looks like you just need to had the test run just after the build as explained [here](https://docs.travis-ci.com/user/languages/cpp#Default-Test-Script)

**Comment by matty0ung** [comment]
Think we can archive this discussion now.
